mindmeld/components/role_classifier.py

```diff
@@ -98,24 +98,26 @@ def fit(self, queries=None, label_set=None, incremental_timestamp=None, **kwargs
             label_set = model_config.train_label_set
             label_set = label_set if label_set else DEFAULT_TRAIN_SET_REGEX
 
-        new_hash = self._get_model_hash(model_config, queries, label_set)
+        new_hash = self._get_model_hash(model_config, label_set)
         cached_model = self._resource_loader.hash_to_model_path.get(new_hash)
 
         if incremental_timestamp and cached_model:
-            logger.info("No need to fit. Loading previous model.")
-            self.load(cached_model)
-            return
+            logger.info("No need to fit. Previous model is cached.")
+            if load_cached:
+                self.load(cached_model)
```

Comment:
Shouldn't we set the following here?
```
        self.ready = True
        self.dirty = True
```

Likewise for the other classifiers where we do:
```
            if load_cached:
                self.load(cached_model)
                return True
```

Reply:
Right, however, I don't know that we should set `dirty = True`.  There are no changes that require saving back to disk, right?

Reply:
Understood, makes sense!

Reply:
My understanding of `dirty` is "This model in memory has changes made to it that should be written to disk".  That's not true if we've just loaded the model from disk.  I may not understand what exactly the `dirty` flag is used for here though.

Reply:
Make sense, no changed needed here